pkg/services/blockfetcher/blockfetcher.go

```diff
@@ -0,0 +1,203 @@
+package blockfetcher
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/url"
+	"time"
+
+	cli "github.com/nspcc-dev/neo-go/cli/server"
+	"github.com/nspcc-dev/neo-go/pkg/core"
+	"github.com/nspcc-dev/neo-go/pkg/core/block"
+	"github.com/nspcc-dev/neo-go/pkg/core/chaindump"
+	"github.com/nspcc-dev/neo-go/pkg/core/storage"
+	"github.com/nspcc-dev/neo-go/pkg/crypto/keys"
+	gio "github.com/nspcc-dev/neo-go/pkg/io"
+	"github.com/nspcc-dev/neo-go/pkg/services/oracle/neofs"
+	"github.com/nspcc-dev/neofs-sdk-go/client"
+	cid "github.com/nspcc-dev/neofs-sdk-go/container/id"
+	"github.com/nspcc-dev/neofs-sdk-go/object"
+	oid "github.com/nspcc-dev/neofs-sdk-go/object/id"
+	"go.uber.org/zap"
+)
+
+type Service struct {
+	chain       *core.Blockchain
+	log         *zap.Logger
+	client      *client.Client
+	db          storage.Store
+	quit        chan bool
+	containerID cid.ID
+	Timeout     time.Duration
+	Nodes       []string
+	dumpDir     string
+}
+
+// ConfigObject represents the configuration object in NeoFS.
+type ConfigObject struct {
+	HashOID   string   `json:"hash_oid"`
+	BlockOIDs []string `json:"block_oid"`
+	Height    uint32   `json:"height"`
+	Timestamp int64    `json:"timestamp"`
+	Step      uint32   `json:"step"`
+}
+
+// Config is the configuration of the BlockFetcherService.
+type Config struct {
+	ContainerID string
+	Nodes       []string
+	Timeout     time.Duration
+	DumpDir     string
+}
+
+// New creates a new BlockFetcherService.
+func New(chain *core.Blockchain, store storage.Store, cfg Config, logger *zap.Logger) *Service {
+	neofsClient, err := client.New(client.PrmInit{})
+	if err != nil {
+		logger.Error("Failed to create NeoFS client", zap.Error(err))
+		return nil
+	}
+	var containerID cid.ID
+	err = containerID.DecodeString(cfg.ContainerID)
+	if err != nil {
+		logger.Error("Failed to decode container ID", zap.Error(err))
+		return nil
+	}
+	return &Service{
+		chain:       chain,
+		log:         logger,
+		client:      neofsClient,
+		db:          store,
+		quit:        make(chan bool),
+		dumpDir:     cfg.DumpDir,
+		containerID: containerID,
+		Nodes:       cfg.Nodes,
+		Timeout:     cfg.Timeout,
+	}
+}
+
+// Name implements the core.Service interface.
+func (bfs *Service) Name() string {
+	return "BlockFetcherService"
+}
+
+// Start implements the core.Service interface.
+func (bfs *Service) Start() {
+	bfs.log.Info("Starting Block Fetcher Service")
+	err := bfs.fetchData()
```

Comment:
IMO, it should be run in a separate goroutine. Once fetching is finished, a server's callback may be called to let the network server know that block fetching is done and other dependent services may be started.